mantis-common/src/main/java/io/mantisrx/common/metrics/MetricsServer.java

```diff
@@ -53,41 +58,51 @@ public class MetricsServer {
         this.tags = tags;
         mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         mapper.registerModule(new Jdk8Module());
+        this.micrometerRegistry = micrometerRegistry;
     }
 
     private Observable<Measurements> measurements(long timeFrequency) {
+        final MeterRegistry microRegistry = this.micrometerRegistry;
         final MetricsRegistry registry = MetricsRegistry.getInstance();
         return
-                Observable.interval(0, timeFrequency, TimeUnit.SECONDS)
-                        .flatMap(new Func1<Long, Observable<Measurements>>() {
-                            @Override
-                            public Observable<Measurements> call(Long t1) {
-                                long timestamp = System.currentTimeMillis();
-                                List<Measurements> measurements = new ArrayList<>();
-                                for (Metrics metrics : registry.metrics()) {
-                                    Collection<CounterMeasurement> counters = new LinkedList<>();
-                                    Collection<GaugeMeasurement> gauges = new LinkedList<>();
-
-                                    for (Entry<MetricId, Counter> counterEntry : metrics.counters().entrySet()) {
-                                        Counter counter = counterEntry.getValue();
-                                        counters.add(new CounterMeasurement(counterEntry.getKey().metricName(), counter.value()));
-                                    }
-                                    for (Entry<MetricId, Gauge> gaugeEntry : metrics.gauges().entrySet()) {
-                                        gauges.add(new GaugeMeasurement(gaugeEntry.getKey().metricName(), gaugeEntry.getValue().doubleValue()));
-                                    }
-                                    measurements.add(new Measurements(metrics.getMetricGroupId().name(),
-                                            timestamp, counters, gauges, tags));
-                                }
-                                return Observable.from(measurements);
+            Observable.interval(0, timeFrequency, TimeUnit.SECONDS)
+                .flatMap(new Func1<Long, Observable<Measurements>>() {
+                    @Override
+                    public Observable<Measurements> call(Long t1) {
+                        long timestamp = System.currentTimeMillis();
+                        List<Measurements> measurements = new ArrayList<>();
+
+
+                        for (Meter meter: microRegistry.getMeters()) {
+                            Collection<MicrometerMeasurement> micrometers = new LinkedList<>();
+                            micrometers.add(new MicrometerMeasurement(meter.getId().getType(), meter.measure().iterator().next().getValue()));
+                            measurements.add(new Measurements(meter.getId().getName(), timestamp, Collections.emptyList(), Collections.emptyList(), micrometers, tags));
```

Comment:
We are generating a measurement instance for every micrometer metric, even though measurements are supposed to consist of a collection of counters, timers, and other components for a specific group ID. Considering this, is the semantics of having a 1:1 relationship between the measurements instance and micrometer metric correct?

Reply:
Yes, it is correct.